### DISCUSSION TITLE
Split participating organisations by role

### DISCUSSION THREAD

**Discussion by bill-anderson** [devinit/ddw-analyst-ui#389] [Ideas]
You will see in the attached file that it is possible to analyse US activities far better if we are able to access single value participating orgs. In this case the US neatly specifies one of each of the participating-org roles. This is clearly not always the case ( many publishers include multiple orgnaisations) but is there value in creating four separate columns, for each role, during ETL?

- I would name the columns as per the code list - i.e. Funding Organisations; Accountable Organisations; Extending Organisations; Implementing Organisations. (NB some publishers use org-ids but too few to standardise on ids)
- I would populate these with Participating Organisation Narrative
- I would delimit multiple values with ; rather than |

Thoughts?

[US-USAGOV Participating Orgs.xlsx](https://github.com/devinit/ddw-analyst-ui/files/5788533/US-USAGOV.Participating.Orgs.xlsx)

**Comment by akmiller01** [comment]
I can't see any technical issues with dividing participating orgs into columns by role, and then separating by ; instead of |.

On the point about populating with narrative, is `narrative` in use more than `@ref`? And should we have a default fallback to `@ref` in the event that narrative is empty?

**Reply by bill-anderson** [reply]
narrative definitely in greater use.

**Comment by k8hughes** [comment]
Any actions from this discussion?  If so it might be good if we follow up ideas with linked git issues so we can see the outcome of discussions.

**Comment by akmiller01** [comment]
Made an issue here: https://github.com/devinit/ddw-analyst-ui/issues/395

**Comment by akmiller01** [comment]
Hey @bill-anderson , implemented this in PR https://github.com/devinit/ddw-analyst-ui/pull/396 but is there any particular reason why you want the names to be separated by semi-colons? I believe I picked pipes because they're relatively rare in the text. There are 26,641 rows in the database presently where participating organisation narrative has a semi-colon in it natively:
```
analyst_ui=# select count(participating_org_narrative) from repo.iati_transactions where participating_org_narrative ILIKE '%;%';
 count 
-------
 26641
(1 row)
```

**Reply by akmiller01** [reply]
Some examples. It seems like some publishers are using semi-colons to distinguish multiple organisations within a single participating-org element anyway, but some are incidental:

- AL MASSAR FOR NOMADS DEVELOPMENT amp; ENVIRONMENT CONSERVATION
- Direction Générale de l’Hydraulique (DGH); Ministère de l'Economie et des Finances
- Ministère de l’Agriculture et des Ressources Animales; Ministère de l’Enseignement Technique et de la Formation Professionnelle, de l’Emploi et du Travail METFPET
- Safi Sana Ghana Limited (SSGL); Training Research and Networkiing for Development - Trend Group

**Reply by akmiller01** [reply]
@bill-anderson did you see the examples above? Can we continue to use a `|` separator to avoid issues?

**Reply by akmiller01** [reply]
I had made an issue and PR for this refactor, it's just awaiting a decision on the separator: https://github.com/devinit/ddw-analyst-ui/issues/395

**Reply by bill-anderson** [reply]
I thought a semi-colon was easier for humans to read.
